code/Proxy/ConfigManifestProxy.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace LeKoala\DebugBar\Proxy;
+
+use SilverStripe\Config\Collections\CachedConfigCollection;
+
+class ConfigManifestProxy extends CachedConfigCollection
+{
+    /**
+     * @var CachedConfigCollection
+     */
+    protected $parent;
+
+    /**
+     * @var array
+     */
+    protected static $configCalls = [];
+
+    /**
+     * @param ConfigCollectionInterface $parent
+     */
+    public function __construct(CachedConfigCollection $parent)
```

Comment:
I'm not sure TBH. The idea is that instead of `$config = new CachedConfigCollection;` you would use the proxy instead: `$config = new ConfigManifestProxy(new CachedConfigCollection)`. Since it extends CachedConfigCollection directly, I think it also makes sense to be this strict about the class it's proxying in the constructor - what do you think?